### PR TITLE
Migrate removed_from_channel_modal to typescript

### PR DESCRIPTION
**Pull request by sowmiyamuthuraman**
Fixes https://github.com/mattermost/mattermost-server/issues/13724

JIRA: https://mattermost.atlassian.net/browse/MM-20516

### CODE REVIEW

File: components/removed_from_channel_modal/removed_from_channel_modal.tsx [hunk 1]
```diff
@@ -2,19 +2,22 @@
...
-    static propTypes = {
-        currentUserId: PropTypes.string.isRequired,
-        onHide: PropTypes.func.isRequired,
-        channelName: PropTypes.string,
-        remover: PropTypes.string,
-    };
+type Props = {
+    currentUserId: string;
+    onHide: () => void;
+    channelName: string;
```
Comment (marianunez) [inline components/removed_from_channel_modal/removed_from_channel_modal.tsx]: We should add optional here given that `channelName` is not required

```suggestion
    channelName?: string;
```

File: components/removed_from_channel_modal/removed_from_channel_modal.tsx [hunk 2]
```diff
@@ -2,19 +2,22 @@
...
+    remover: string;
```
Comment (marianunez) [inline components/removed_from_channel_modal/removed_from_channel_modal.tsx]: Same here
```suggestion
    remover?: string;
```

**Comment by hanzei** [comment]
Thanks for the contribution @sowmiyamuthuraman :+1:

**Pull request merged**